.github/workflows/docker-image.yml

```diff
@@ -0,0 +1,29 @@
+name: Docker Image CI
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+jobs:
+
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v3
+    - name: Run the build.sh scripts
+      env:
+        DOCKER_HUB_TOKEN: ${{ secrets.DOCKER_HUB_RW_TOKEN }}
+      run: |
+            if [ -n "$DOCKER_HUB_TOKEN" ]
+            then 
+              echo "Secrets detected, can't run build script"
+              exit 1 # terminate and indicate error
+            else 
+              # find the build.sh scripts and execute them
+              find . -iname 'build.sh' -exec ./{} \;
+            fi
```

Comment:
I'm not sure this is necessary; GitHub doesn't forward secrets into CI jobs dispatched from forks.

So, the only case in which the DOCKER_HUB_TOKEN would be populated is if one of us maintainers pushes a branch directly to this repo.

Reply:
Well, the check itself won't matter, because the PR could alter the GA workflow.

What we wanted was to check that it was also not making the secret available to other repos.

Reply:
So, you can turn the step into:
````yaml
    - name: Run the build.sh scripts
      run: |
              find . -iname 'build.sh' -exec ./{} \;
```

Reply:
Probably need to change from `find -exec` to `find | xargs`

Reply:
oh, interesting, GNU find (AKA what you have in Ubuntu) apparently has a `-quit` argument?